src/main/java/org/prebid/server/bidder/amx/AmxBidder.java

```diff
@@ -0,0 +1,264 @@
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * AMX {@link Bidder} implementation.
+ */
+public class AmxBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpAmx>> AMX_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpAmx>>() {
+            };
+
+    private static final String ADAPTER_VERSION = "pbs1.0";
+    private static final String VERSION_PARAM = "v";
+    private static final String VAST_SEARCH_POINT = "</Impression>";
+    private static final String VAST_IMPRESSION_FORMAT = "<Impression><![CDATA[%s]]></Impression>";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public AmxBidder(String endpointUrl, JacksonMapper mapper) {
+        this.mapper = Objects.requireNonNull(mapper);
+        this.endpointUrl = new URIBuilder()
+                .setPath(HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl)))
+                .addParameter(VERSION_PARAM, ADAPTER_VERSION)
+                .toString();
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<Imp> modifiedImps = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+        String publisherId = null;
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpAmx extImpAmx = parseImpExt(imp);
+                final String tagId = extImpAmx.getTagId();
+                if (StringUtils.isNotBlank(tagId)) {
+                    publisherId = tagId;
+                }
+
+                final String adUnitId = extImpAmx.getAdUnitId();
+                if (StringUtils.isNotBlank(adUnitId)) {
+                    modifiedImps.add(imp.toBuilder().tagid(adUnitId).build());
+                }
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        final BidRequest outgoingRequest = createOutgoingRequest(request, publisherId, modifiedImps);
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<BidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(endpointUrl)
+                        .headers(headers())
+                        .payload(outgoingRequest)
+                        .body(mapper.encode(outgoingRequest))
+                        .build()), errors);
+    }
+
+    private ExtImpAmx parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), AMX_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private BidRequest createOutgoingRequest(BidRequest request, String publisherId, List<Imp> imps) {
+        return updateRequestIfPublisherIdPresent(request, publisherId).toBuilder()
```

Comment:
It is too expensive to create request-builder twice here. 
Pls think how we can optimize.